packages/amplify-headless-interface/scripts/generateSchemas.ts

```diff
@@ -0,0 +1,49 @@
+import { getProgramFromFiles, buildGenerator, PartialArgs, Definition } from 'typescript-json-schema';
+import fs from 'fs-extra';
+import path from 'path';
+
+// paths are relative to the package root
+const typesSourceRoot = './src/interface/';
+const schemaFilesRoot = './schemas';
+
+// defines the type names and the paths to the TS files that define them
+const typeDefs: TypeDef[] = [
+  {
+    typeName: 'AddAuthRequest',
+    relativeSourcePaths: ['addAuth.ts'],
+  },
+  {
+    typeName: 'AddApiRequest',
+    relativeSourcePaths: ['addApi.ts'],
+  },
+  {
+    typeName: 'AddStorageRequest',
+    relativeSourcePaths: ['addStorage.ts'],
+  },
+];
+
+const schemaFileName = (typeName: string) => `${typeName}.schema.json`;
+const forceFlag = '--overwrite';
```

Comment:
Are we auto-bumping the schema version of overwrite or force flags aren't present?

Reply:
The schema contains the version number and this script will pick that up and save the schema in the correct versioned folder. This flag is if you are trying to modify an existing schema version (which should only happen when working on a new version and you need to iterate on it as you're working)